integration-tests/src/lib.rs

```diff
@@ -92,3 +96,97 @@ macro_rules! generate_abi_fn {
         }
     }};
 }
+
+pub struct BuildResult {
+    pub wasm: Vec<u8>,
+    pub abi_root: Option<near_abi::AbiRoot>,
+    pub abi_compressed: Option<Vec<u8>>,
+}
+
+// TODO: make cargo-near agnostic of stdin/stdout and capture the resulting paths from Writer
+#[macro_export]
+macro_rules! build_with {
+    ($(Cargo: $cargo_path:expr;)? $(Vars: $cargo_vars:expr;)? $(Opts: $cli_opts:expr;)? Code: $($code:tt)*) => {{
+        let opts = "cargo near build";
+        $(let opts = format!("cargo near build {}", $cli_opts);)?;
+        let result_dir = $crate::invoke_cargo_near! {
+            $(Cargo: $cargo_path;)? $(Vars: $cargo_vars;)?
+            Opts: opts;
+            Code:
+            $($code)*
+        };
+
+        let manifest_dir: std::path::PathBuf = env!("CARGO_MANIFEST_DIR").into();
+        let workspace_dir = manifest_dir.parent().unwrap().join("target").join("_abi-integration-tests");
```

Comment:
\+ `std::env::set_var("CARGO_TARGET_DIR", workspace_dir.join("target"));`

Reply:
Why do I need this here? This snippet is needed purely to decide where the resulting wasm file is located.

Reply:
Ah, right, `build_with!` invokes `invoke_cargo_near!`. Ignore this.